source/slang/slang-check-decl.cpp

```diff
@@ -2574,6 +2574,135 @@ namespace Slang
         return false;
     }
 
+    GenericDecl* SemanticsVisitor::synthesizeGenericSignatureForRequirementWitness(
```

Comment:
Are there any actual changes made to this function?

Reply:
This is the generic version of the signature synthesis method.
Creates the generic wrapper, clones the parameters & constraints, then calls method synthesis with substituted parameters in place

Reply:
Ah, I see. Seems like there are code copied from an existing function, is there a way to bottleneck them through the same function?

Reply:
I'm not sure it's possible.. the flow is similar (create empty decl, then clone parameters), but the rest is different (type params need to be explicitly substituted into the constraints, as well as into the decl-ref so that the inner function uses the new type params for its declarations, etc..)